### ISSUE TITLE
Using "Restore" may result in Death

### ISSUE THREAD

**Issue by KellenM** [HabitRPG/habitica#1600]
I noticed recently that when you enter the "Restore" dialog, changes are made to attributes _as you type_, which sort of contradicts the "Save" part of the "Save & Close" button that closes the dialog.

This some people highlight and start typing over, which bypasses this issue, but some people delete the entire text and re-type it. This makes a big difference, due to the "real time" updating.

Say I have 47.2 gold and I'm trying to get it to have 35.1 gold. I click so the cursor is at the far right of the 47.2 text, and then delete, using backspace, and then type in the number I want. As I type it performs the following actions, keystroke-by-keystroke: 
1. \- 0.2 gold
2. (nothing, I deleted a decimal point)
3. \- 43 gold
4. \- 4 gold
5. \+ 3 gold
6. \+ 32 gold
7. (nothing, adding the decimal point)
8. \+ 0.1 gold

Not bad, everything seems fine. Odd that it updates at every keystroke, instead of just when I click save but who cares? Well, if we do that exact same thing, with the HP instead, something terrible happens right around the fourth keystroke. Namely, death.

I think the Restore dialog needs to not update stats and things in real-time. Just when the user actually hits "Save & Close."

**Comment by Shaners** [comment]
The restore feature is going away eventually so I'm not sure this type of fix is going to be a priority.

**Comment by KellenM** [comment]
Oh! I didn't know that. In that case, I would agree that it's not a priority at all. But if any type of "type in your attribute here" remains somewhere, it would be good to hold changes until users hit "submit."

**Comment by Shaners** [comment]
Yea agreed, I think everywhere else waits for a save or + button to be pressed first.

**Comment by octern** [comment]
This is a very clear and entertaining bug report. Also, I'm in favor of keeping the restore feature. I mostly use it to take away xp if I think I got too much for a task.

**Comment by paglias** [comment]
@lefnire what do you think about that? I have a fix ready for this so that values are saved only after clicking "Save".

I would like to use the same system also for profile editing, what do you think?

**Comment by lefnire** [comment]
Sounds good re: restore saving. I thought it was already that way, guess I was wrong - yeah, pull the trigger.

Regarding profile-editing, the only difficulty is going to be the adding / deleting websites feature. We could just apply it to everything except websites. We could also remove the websites field altogether, since we now support Markdown and they can add arbitrary lists (including for websites)

**Comment by paglias** [comment]
@lefnire I've applied this also to websites, the commits are https://github.com/lefnire/habitrpg/commit/68426c7ccbe46c3be6ca439d6918ce0bd1a4c400 for profile editing and https://github.com/lefnire/habitrpg/commit/4e56df0e8241e8c5a15bb21528f8c912a7f9e8a0 for restore values.
